include/nbl/builtin/hlsl/tgmath/impl.hlsl

```diff
@@ -240,7 +247,13 @@ struct beta_helper<T NBL_PARTIAL_REQ_BOT(concepts::FloatingPointScalar<T>) >
 	// implementation from Numerical Recipes in C, 2nd ed.
 	static T __call(T v1, T v2)
 	{
-		return exp_helper<T>::__call(lgamma_helper<T>::__call(v1) + lgamma_helper<T>::__call(v2) - lgamma_helper<T>::__call(v1+v2));
+		// specialized for Cook Torrance BTDFs
+		assert(v1 >= 1.0 && v2 >= 1.0);
+
+		if (v1+v2 > 1e6)
+			return T(0.0);
+
+		return exp2_helper<T>::__call(l2gamma_helper<T>::__call(v1) + l2gamma_helper<T>::__call(v2) - l2gamma_helper<T>::__call(v1+v2));
```

Comment:
if you want to use l2gamma for beta, you need a non-stirling implementation for float64 as well

P.S. its best if `lgamma` itself calls `l2gamma` and rescales its output (the mul by cosntant will get constant folded with other pats of the expression)

Reply:
you need only non-stirling for l2gamma anyway